File: src/Java/169.MajorityElement.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> m = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (m.containsKey(nums[i]) == false) {
                m.put(nums[i], 1);
            }
            else{
                int val = m.get(nums[i]);
                m.remove(nums[i]);
                m.put(nums[i], val + 1);
            }
        }
        Collection<Integer> s = m.values();
        int majar = Collections.max(s);
        for (Map.Entry<Integer, Integer> en : m.entrySet()) {
            if (en.getValue() == majar) {
                return en.getKey();
            }
        }
        return -1;
    }
}

File: src/Python/101-200/169.MajorityElement.py

class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        if n <= 2:
            return nums[0]
        d = {}
        for i in range(n):
            if nums[i] not in d.keys():
                d[nums[i]] = 1
            else:
                val = d[nums[i]]
                d[nums[i]] = val + 1
        majar = max(d.values())
        for (u, v) in d.items():
            if v == majar:
                return u

if __name__ == '__main__':
    solution = Solution()
    print(solution.majorityElement([2, 2]))
    print(solution.majorityElement([1, 2, 2, 3]))
else:
    pass
